### PR TITLE
Add CircleCI config that builds and lints the code

### PR DESCRIPTION
**Pull request by mythmon**
As well as getting running, this PR also makes the lints pass.

### CODE REVIEW

**Comment by mythmon** [comment]
To get Circle CI running, it should be enough for someone with write access to this repo to set up the repository at https://circleci.com/add-projects/gh/mozilla

**Comment by nhi-nguyen** [comment]
thanks for doing this @mythmon! I added you to the repo. I/Maxx can set up the circleci project later (if you don't get to it first :-) )

**Comment by mythmon** [comment]
Apparently I was wrong about the requirements to set up a project, and a repo admin will have to do it. I don't think I need to be a repo admin here, so I'll leave it to one of you two to do.

File: src/background.js [hunk 1]
```diff
@@ -2,7 +2,9 @@
 /* global browser, runHeuristics */
 
 function log() {
-  if (false) {
+  // eslint-disable-next-line no-constant-condition
+  if (true) {
```
Comment (jonathanKingston) [inline src/background.js]: Please disable this again.

File: src/experiments/heuristics/api.js [hunk 2]
```diff
@@ -5,7 +5,9 @@ let Cu3 = Components.utils;
 Cu3.import("resource://gre/modules/Services.jsm");
 
 function log() {
-  if (false) {
+  // eslint-disable-next-line no-constant-condition
+  if (true) {
```
Comment (jonathanKingston) [inline src/experiments/heuristics/api.js]: Disable this again too

File: src/background.js (hunk 1 above)
Comment (mythmon) [reply src/background.js]: Oops, sorry about that.

**Comment by jonathanKingston** [comment]
I have tried to get circle before setup on a Mozilla repo, I'm not sure the current steps to enablement, even as an admin I couldn't get it working.

Patch looks great, minus the two nits. Feel free to make a global variable or whatever if it makes it easier to control debugging.

**Comment by maxxcrawford** [comment]
> I have tried to get circle before setup on a Mozilla repo, I'm not sure the current steps to enablement, even as an admin I couldn't get it working.
> 
> Patch looks great, minus the two nits. Feel free to make a global variable or whatever if it makes it easier to control debugging.

Yes, please! 🙏

**Comment by maxxcrawford** [comment]
@mythmon Can you catch this up to the latest `master`?

File: src/background.js [hunk 3]
```diff
@@ -76,6 +78,7 @@ const stateManager = {
       log("Mismatched, curMode: ", curMode);
       if (curMode === 0 || curMode === 5) {
         // If user has manually set trr.mode to 0, and it was previously something else.
+        // eslint-disable-next-line no-undef
         browser.experiments.heuristics.sendHeuristicsPing("userModified", results);
```
Comment (mythmon) [inline src/background.js]: I think that this is a bug that is pre-existing. I'm not sure how to fix it.

File: src/background.js [hunk 4]
```diff
@@ -137,12 +139,13 @@ const rollout = {
     await stateManager.setState("UIDisabled");
     await stateManager.rememberDoorhangerDecision("UIDisabled");
     await stateManager.rememberDoorhangerPingSent();
+    // eslint-disable-next-line no-undef
     browser.experiments.heuristics.sendHeuristicsPing("doorhangerDecline", results);
```
Comment (mythmon) [inline src/background.js]: I think that this is a bug that is pre-existing. I'm not sure how to fix it.

**Comment by mythmon** [comment]
> > Feel free to make a global variable or whatever if it makes it easier to control debugging.
> 
> Yes, please! :pray:

I didn't end up doing this because I'm not sure how what the best way to introduce global variables is. I'm just trying to set up CI :joy:.

File: src/background.js (hunk 4 above)
Comment (maxxcrawford) [reply src/background.js]: @mythmon Confirmed. I'll fix this before signing v10

**Comment by nhi-nguyen** [comment]
I've set up the CircleCI project and looks like things are working 🎉 : https://circleci.com/gh/mozilla/doh-rollout/tree/master

**Pull request merged**